### PR TITLE
Implemented Drag and Drop Feature to DetailsViewController

### PR DESCRIPTION
**Pull request by amanial1234**
- Created DetailsViewController for when the Cell in MatchmakingViewController is pressed
- Created prepareforSegue in MatchMakingViewController to pass the User Information to DetailsViewController
- Created properties for DetailsViewController such as screenName and profileView from the User
- Created PFUser property Author that stores the User 
- Created Outlets for the drag and drop such as the PanGesture, the accept and rejectViews and Labels, the AreaView, the DragView, and outlets for the X and Y constraints for the DragView
- Created properties for the initial DragView position, a bool if either the accept View or reject View is reached, and the bounds
- Created MoveObject Function that allows the view to move using constraints once the panGesture is recognized
- Set the user's name in DetailsViewController

TESTING

Test Case 1: Used the build simulator to test if the accept or reject views are detects by changing the color and text of the view
Test Case 2: Used Breakpoints to make sure the gesture was called upon correctly
Test Case 3: Made sure the name was displayed on the DetailsViewController using the build Simulator

Uploading Screen Recording 2022-07-30 at 7.14.47 PM.mov…

### CODE REVIEW

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^CompletionBlock)();
+
+@interface DetailsViewController : UIViewController
+
+@property (copy, nonatomic) CompletionBlock completion;
+
+@property (nonatomic, strong) PFUser *author;
```
Comment (amanial1234) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.h:12]: object of the user we are currently viewing

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@property (weak, nonatomic) IBOutlet UIImageView *profileView;
+
+
+@property (weak, nonatomic) IBOutlet UIView *dragAreaView;
+@property (weak, nonatomic) IBOutlet UIView *dragView;
+@property (weak, nonatomic) IBOutlet UIView *acceptView;
+@property (weak, nonatomic) IBOutlet UIView *rejectView;
+@property (weak, nonatomic) IBOutlet UILabel *acceptLabel;
+@property (weak, nonatomic) IBOutlet UILabel *rejectLabel;
+@property (weak, nonatomic) IBOutlet NSLayoutConstraint *dragViewX;
```
Comment (amanial1234) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.h:23]: X and Y constraints of the dragView

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+    self.acceptView.layer.cornerRadius = self.acceptView.bounds.size.height / 2;
+    self.acceptView.layer.borderWidth = 5;
+
+    self.rejectView.layer.cornerRadius = self.rejectView.bounds.size.height / 2;
+    self.rejectView.layer.borderWidth = 5;
+    
+    self.initialDragViewY = self.dragViewY.constant;
+    
+    [self updateAcceptView];
```
Comment (amanial1234) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:32]: Updates the accept and Reject View

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    }
+    else if (self.panGesture.state == UIGestureRecognizerStateEnded) {
+        if (self.isAcceptReached) {
+            [self.navigationController popToRootViewControllerAnimated:YES];
+            if (self.completion) {
+                self.completion();
+            }
+        }
+        if (self.isRejectReached) {
+            [self.navigationController popToRootViewControllerAnimated:YES];
```
Comment (amanial1234) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:70]: Transitions into rootviewController if Rejected or Accepted is confirmed

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    } completion:nil];
+    
+    [self updateAcceptView];
+    [self updateRejectView];
+    
+}
+
+- (void)updateAcceptView {
+    //Updates the Accept view to a green if it is reached and changes text
+    UIColor *goalColor = self.isAcceptReached ? [UIColor greenColor] : [UIColor colorWithRed:174/255.0 green:0 blue:0 alpha:1];
```
Comment (amanial1234) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:136]: Turns Green and changes text to Accepted if view is in Accept View

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    
+    self.acceptView.layer.borderColor = goalColor.CGColor;
+    
+    self.acceptLabel.textColor = goalColor;
+    self.acceptLabel.text = self.isAcceptReached ? @"Accepted!" : @"Accept!";
+}
+- (void)updateRejectView {
+    //Updates the Reject view to a brighter red if it is reached and changes text
+    UIColor *goalColor = self.isRejectReached ? [UIColor redColor] : [UIColor colorWithRed:174/255.0 green:0 blue:0 alpha:1];
```
Comment (amanial1234) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:145]: Turns Red and changes text to Rejected if view is in Accept View

File: MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m [hunk 7]
```diff
@@ -81,5 +82,14 @@ -(void)getMatchesDictionary{
     }];
 }
 
-
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
```
Comment (amanial1234) [inline MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m:85]: Added prepareforsegue to transition information of the User to detailsViewController

**Review by amanial1234 — COMMENTED** [review]
Decided to do the Drag and Drop in a DetailsViewController but I have accomplished implementing it in a Cell as well but found it to take up to much space and didn't look as polished

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.h (hunk 1 above)
Comment (nathanielcook) [reply MusicTaste/MusicTaste/View Controllers/DetailsViewController.h:12]: I remember you added a model class (User.h and User.m) previously. Is that still in use?

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    self.rejectView.layer.borderWidth = 5;
+    
+    self.initialDragViewY = self.dragViewY.constant;
+    
+    [self updateAcceptView];
+    [self updateRejectView];
+
+    
+}
+- (void)viewDidLayoutSubviews {
```
Comment (nathanielcook) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:37]: Try to have consistency for amount of blank lines between functions

```
    [self updateRejectView];
}

- (void)viewDidLayoutSubviews
```

**Comment by nathanielcook** [comment]
> Decided to do the Drag and Drop in a DetailsViewController but I have accomplished implementing it in a Cell as well but found it to take up to much space and didn't look as polished

Great! Did you save the sample code? Would have been great for that to be two separate PRs (one for it in the cell and one for it in DetailsViewController). What was the key to making it work in the cell?

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+
+- (void)pan:(UIPanGestureRecognizer *)gesture
+{
+    NSLog(@"testing");
+}
+
+#pragma mark -
+
+
+@end
```
Comment (nathanielcook) [inline MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:186]: Remove unused pragmas, unused functions and logging that is no longer necessary

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m (hunk 6 above)
Comment (nathanielcook) [reply MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:145]: Cool!

File: MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m [hunk 10]
```diff
@@ -81,5 +82,14 @@ -(void)getMatchesDictionary{
...
+    //Segue to Details View Controller
+    if([[segue identifier] isEqualToString:@"detailsSegue"]) {
+        UITableViewCell *cell = sender;
+        NSIndexPath *indexpath = [self.MatchTableView indexPathForCell:cell];
```
Comment (nathanielcook) [inline MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m:89]: Don't forget about camelCase

`NSIndexPath *indexPath = [self.MatchTableView indexPathForCell:cell];`

**Review by nathanielcook — COMMENTED** [review]
Happy you figured this out. Would you mind re-attaching your demo video? It seems like it didn't get fully attached.

**Comment by amanial1234** [comment]
https://user-images.githubusercontent.com/103143506/182269218-ca951ab1-d3bb-4e5c-83c0-d7b736ccc179.mov

**Comment by amanial1234** [comment]
> 

I have a PR of the Drag and Drop of the View called Test Drag and Drop on my github

File: MusicTaste/MusicTaste/View Controllers/DetailsViewController.m (hunk 9 above)
Comment (amanial1234) [reply MusicTaste/MusicTaste/View Controllers/DetailsViewController.m:186]: I will change that right now!

File: MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m (hunk 10 above)
Comment (amanial1234) [reply MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m:89]: Will do!

**Pull request merged**